src/nbl/video/CVulkanCommandPool.h

```diff
@@ -0,0 +1,90 @@
+#ifndef __NBL_C_VULKAN_COMMAND_POOL_H_INCLUDED__
+
+#include "nbl/video/IGPUCommandPool.h"
+#include "nbl/core/containers/CMemoryPool.h"
+
+#include <mutex>
+
+#include <volk.h>
+
+namespace nbl::video
+{
+
+struct ArgumentReferenceSegment
```

Comment:
Next the struct inside `CVulkanCommandPool` (dont want it floating around the video namespace)